docs/advanced-usage/define-causer-for-runtime.md

```diff
@@ -0,0 +1,29 @@
+---
+title: Define causer for runtime
+weight: 4
+---
+
+In many cases you may want to set causer globally maybe inside jobs where there's no logged-in user, v4 made this possible by introducing `CauserResolver` that will allow you to set the causer globally. See the example:
+
+```php
+// in a queue job or controller
+
+use Spatie\Activitylog\Facade\CauserResolver;
+
+// ... other code
+
+$product = Product::first(1);
+$causer = $product->owner;
+
+CauserResover::setCauser($causer);
```

Comment:
`CauserResolver` sound very "programmy" to me. 

This might be a bit more "human":

```
Activity::determineCauserUsing($callableOrCauser)
```

Reply:
Like adding new static method on `Activity` class calling `CauserResolver::setCauser()` or rename the `CauserResolver` ?

Reply:
I voted against the method on `Activity` in first place to keep all classes clean and scoped.

Reply:
To me, `CauserResolver` is something that a user will have a hard time remembering. `Activity::determineCauserUsing` just sounds better and more rememberable to me. In general, For me, optimising DX is more important that optimising for clean and scoped classes. 

Not putting any veto here. Probably, you'll spend the time maintaining this, so in that sense, your opinion on this is important too.

Reply:
I tend to agree more with @freekmurze that `Activity::determineCauserUsing` is much better DX than `CauserResolver`.

@Gummibeer Adding this static method on the `Activity` class that will just call  underlying `CauserResolver ::setCauser` with same signature would be a better trade off and also the `Activity` would not know about the causer resolving logic.

Reply:
I will spend some time today to polish/finish everything and prepare or even do the release.
So please everyone don't push anything anymore.^^
Thanks again to @nagi1 for the awesome work! 🚀

Reply:
I trust your judgment. Hope we see v4 soon :)